core/templates/dev/head/pages/exploration_player/ExplorationPlayerStateService.js

```diff
@@ -0,0 +1,143 @@
+      PlayerTranscriptService,
+      ReadOnlyExplorationBackendApiService, PretestQuestionBackendApiService) {
+    var _currentEngineService = null;
+    var _inPretestMode = false;
+    var _editorPreviewMode = ContextService.isInExplorationEditorPage();
+    var _explorationId = ContextService.getExplorationId();
+    var _version = GLOBALS.explorationVersion;
+    var _storyId = UrlService.getStoryIdInPlayer();
+
+    var _initializeExplorationServices = function(
+        returnDict, arePretestsAvailable, callback) {
+      StateClassifierMappingService.init(
+        returnDict.state_classifier_mapping);
+      StatsReportingService.initSession(
+        _explorationId, returnDict.exploration.title,
+        _version, returnDict.session_id, GLOBALS.collectionId);
+      if (ENABLE_PLAYTHROUGH_RECORDING) {
+        PlaythroughService.initSession(explorationId, version);
+      }
+      PlayerCorrectnessFeedbackEnabledService.init(
+        returnDict.correctness_feedback_enabled);
+      ExplorationEngineService.init(
+        returnDict.exploration, returnDict.version,
+        returnDict.preferred_audio_language_code,
+        returnDict.auto_tts_enabled,
+        arePretestsAvailable ?
+        function() {} : callback);
+    };
+
+    var _initializePretestServices = function(pretestQuestionDicts, callback) {
+      PlayerCorrectnessFeedbackEnabledService.init(true);
+      PretestEngineService.init(pretestQuestionDicts, callback);
+    };
+
+    var _setExplorationMode = function() {
+      _inPretestMode = false;
+      _currentEngineService = ExplorationEngineService;
+    };
+
+    var _setPretestMode = function() {
+      _inPretestMode = true;
+      _currentEngineService = PretestEngineService;
+    };
+
+    return {
+      getCurrentEngineService: function() {
+        return _currentEngineService;
+      },
+      isInPretestMode: function() {
+        return _inPretestMode;
+      },
+      getPretestQuestionCount: function() {
+        return PretestEngineService.getPretestQuestionCount();
+      },
+      moveToExploration: function(callback) {
+        _setExplorationMode();
+        ExplorationEngineService.moveToExploration(callback);
+      },
+      initializePlayer: function(callback) {
+        PlayerTranscriptService.init();
+        if (_editorPreviewMode) {
+          _setExplorationMode();
+          EditableExplorationBackendApiService.fetchApplyDraftExploration(
+            explorationId).then(function(returnDict) {
+            ExplorationEngineService.init(
+              returnDict, null, null, null, _initializeDirectiveComponents);
+            PlayerCorrectnessFeedbackEnabledService.init(
+              returnDict.correctness_feedback_enabled);
+            NumberAttemptsService.reset();
+          });
+        } else {
+          if (_version) {
+            $q.all([
+              ReadOnlyExplorationBackendApiService.loadExploration(
+                _explorationId, _version, _storyId),
```

Comment:
loadExploration takes only two args. Should you be passing in storyId here?

Reply:
Done.